crates/rspack_plugin_javascript/src/visitors/mod.rs

```diff
@@ -240,6 +242,7 @@ pub fn run_before_pass(
       builtins_webpack_plugin(options, unresolved_mark),
       // This will be deprecated in the future when TypeScript support and transform by default is dropped.
       // These features are coupled with this.
+      replace_top_level_this(unresolved_mark),
```

Comment:
And move this under this branch https://github.com/web-infra-dev/rspack/blob/3c0c6ca9d9cc5f99919520288006e33f14867e57/crates/rspack_plugin_javascript/src/visitors/dependency/mod.rs#L131